src/coverlet.core/Coverage.cs

```diff
@@ -420,11 +420,15 @@ private void CalculateCoverage()
                         var hitLocation = result.HitCandidates[i];
                         var document = documentsList[hitLocation.docIndex];
                         int hits = br.ReadInt32();
+                        hits = hits < 0 ? int.MaxValue : hits;
 
```

Comment:
No need to do the work below if there were not hits at the current location (which links up with the next suggestion):

```suggestion
if (hits == 0)
{
    continue;
}
```

Reply:
done